src/cascade/model/operations.py

```diff
@@ -4,282 +4,198 @@
 
+     * Random Effects - These are not set because the parent cannot inform
+       the random effect of its grandchildren.
 
-def estimate_priors_from_posterior_draws(draws, model_context, execution_context):
-    r"""
-    Given a dataframe of sampled outcomes from a previous run of DismodAT,
-    find parameters for the given distributions. This works from the model
-    context. The draws are a representation of :math:`p(\theta, u|y,\eta)`
-    where :math:`\eta` are the parameters for the prior distributions on
-    :math:`\theta`, the model variables, :math:`y` are the data,
-    and :math:`u` are the random effects. The goal of this function is to
-    calculate the likelihood of the parameters for the prior distribution
-    of the next step down in the cascade,
+     * Covariate Multipliers - These are the alpha, beta, and gamma,
+       and they will be set from parent's draws.
 
-    .. math::
+    Any prior distributions that are fixed in the model are not set.
+    If the posterior draw is fixed, then it will be fixed in this model.
 
-        p(\eta|y) = \int p(\eta|\theta) p(\theta|y) d\theta
+    Args:
+        model_context (ModelContext): The Model.
+        posterior_draws (DataFrame): This dataframe has multiple copies of
+            the ``fit_var`` table, each with a different fit for all model
+            variables. The draws also contain merged columns from the ``var``
+            table so that we know what the variables are for.
 
-    and use that to generate one value of the most likely priors,
-    :math:`\eta^*` that will initialize the next step down.
+    Returns:
+        None: It changes priors on the model context.
+    """
+    if posterior_draws is None or posterior_draws.empty:
+        return
+    _assign_rate_priors(model_context, posterior_draws)
+    _assign_mulcov_priors(model_context, posterior_draws)
+
+
+def _assign_rate_priors(model_context, posterior_draws):
+    grandparent_id = model_context.parameters.grandparent_location_id
+    parent_id = model_context.parameters.location_id
+    underlying_rate = dict()
+    random_effect = dict()
+    rate_draws = posterior_draws[posterior_draws.covariate_id.isna()]
+    # Posterior draws are copies of the var table, so they have sections
+    # for each random field. The (smooth_id, node_id) uniquely identifies
+    # each random field in the vars table.
+    for unique_field, field_df in rate_draws.groupby(["smooth_id", "location_id"]):
+        traits = field_df.iloc[0]
+        rate_name = RateName(traits.rate_id).name
+        # On the odd chance that the draws are for this location, passed into
+        # itself again, the order of these if-then should check first for
+        # the underlying rate because both grandparent and parent will match.
+        if traits.location_id == grandparent_id:
+            underlying_rate[rate_name] = field_df
+        elif traits.location_id == parent_id:
+            random_effect[rate_name] = field_df
```

Comment:
Ohhhh! You're right. Yes, this is wrong for that case. Nice catch. I'm not going to fix this one because EpiViz-AT can't pass to itself, but I'll fix it in the new version.